python_modules/dagster/dagster/_core/storage/upath_io_manager.py

```diff
@@ -58,6 +58,14 @@ def get_metadata(
     def has_output(self, context: OutputContext) -> bool:
         return self._get_path(context).exists()
 
+    def _with_extension(self, path: UPath) -> UPath:
+        # Can't just call path.with_suffix(self.extension) because if
```

Comment:
In the subset of cases where `extension != ""`, I think that, even after this change, the problem you raised will still exist.

StackOverflow has a suggestion on how to fix this: https://stackoverflow.com/a/53092299/6258015. Thoughts?

Reply:
Agreed. I am fine with any of the methods in that SO link. I will pick one and push.

Reply:
Actually a further question @sryza. Suppose the path ends with "/path/to/file.pickle" and extension is "pickle". Should the result be "/path/to/file.pickle.pickle"?

Reply:
> Actually a further question @sryza. Suppose the path ends with "/path/to/file.pickle" and extension is "pickle". Should the result be "/path/to/file.pickle.pickle"?

I think yes, just for consistency.